### PR TITLE
[Frontend] Fix AutoGraph not converting functions wrapped by 'adjoint', 'grad', etc.

### PR DESCRIPTION
**Pull request by dime10**
The following example would fail before the fix: 
```py
 def inner(n):
    for i in range(n):
      qml.T(i)

  @qjit(autograph=True)
  @qml.qnode(qml.device("lightning.qubit", wires=5))
  def f(n: int):
      adjoint(inner)(n)
      return qml.state()
```

### CODE REVIEW

File: frontend/catalyst/ag_primitives.py [hunk 1]
```diff
@@ -389,6 +389,27 @@ def converted_call(fn, args, kwargs, caller_fn_scope=None, options=None):
...
+            catalyst.ctrl,
+            catalyst.grad,
+            catalyst.jacobian,
+            catalyst.vjp,
+            catalyst.jvp,
+        ):
+            assert args and callable(args[0])
+            wrapped_fn = args[0]
+
+            def passtrough_wrapper(*args, **kwargs):
```
Comment (erick-xanadu) [inline frontend/catalyst/ag_primitives.py:404]: typo -> `passthrough_wrapper`

Comment (erick-xanadu) [reply frontend/catalyst/ag_primitives.py:404]:
```suggestion
           def passthrough_wrapper(*args, **kwargs):
```

File: frontend/catalyst/ag_primitives.py [hunk 2]
```diff
@@ -389,6 +389,27 @@ def converted_call(fn, args, kwargs, caller_fn_scope=None, options=None):
...
+                return converted_call(wrapped_fn, args, kwargs, caller_fn_scope, options)
+
+            return fn(
+                passtrough_wrapper,
```
Comment (erick-xanadu) [inline frontend/catalyst/ag_primitives.py:408]:
```suggestion
                passthrough_wrapper,
```

**Review by erick-xanadu — CHANGES_REQUESTED** [review]
I am not an expert in autograph yet. The tests look good. I have noticed that some tests use the variable backend while others are hard coded to be lightning. I think it is best if we always use backend.

**Comment by dime10** [comment]
> The tests look good. I have noticed that some tests use the variable backend while others are hard coded to be lightning. I think it is best if we always use backend.

I deliberately didn't use it here because I'm only testing frontend functionality. I just had to use a QNode to use adjoint, otherwise I wouldn't have. I agree that most tests should use backend, unless they are not testing execution functionality.

**Review by erick-xanadu — APPROVED** [review]

**Pull request merged**